rust_src/src/character.rs

```diff
@@ -26,6 +27,18 @@ pub const fn char_head_p(byte: c_uchar) -> bool {
     (byte & 0xC0) != 0x80
 }
 
+/// Return width of CHAR when displayed in the current buffer.
+/// The width is measured by how many columns it occupies on the screen.
+/// Tab is taken to occupy `tab-width' columns.
+/// usage: (char-width CHAR)
+#[lisp_fn(c_name = "char_width", name = "char-width")]
+pub fn char_width_lisp(ch: LispObject) -> EmacsInt {
```

Comment:
The `lisp_fn` macro can check and convert arguments for us if you let it. Specify that the input type is an `EmacsInt` here and it will convert it from `LispObject` for you, and bail out with an error if someone passes in anything but a number.